Fatras/include/ActsFatras/Digitization/Channelizer.hpp

```diff
@@ -8,97 +8,66 @@
 
 #pragma once
 
-#include "Acts/Definitions/Algebra.hpp"
-#include "Acts/Geometry/GeometryContext.hpp"
+#include "ActsFatras/Digitization/PlanarSurfaceDrift.hpp"
+#include "ActsFatras/Digitization/PlanarSurfaceMask.hpp"
+#include "ActsFatras/Digitization/Segmentizer.hpp"
+#include "ActsFatras/EventData/Hit.hpp"
 
-#include <array>
-#include <utility>
-#include <vector>
-
-namespace Acts {
-class BinUtility;
-class Surface;
-}  // namespace Acts
+#include <numeric>
 
 namespace ActsFatras {
 
-/// The Channelizer splits a surface segment, i.e. after projection
-/// onto the readout surface into channel segments.
-///
-struct Channelizer {
-  /// Shorthand for a 2D segment
-  using Segment2D = std::array<Acts::Vector2, 2>;
-  /// Shorthand for a 2D bin
-  using Bin2D = std::array<unsigned int, 2>;
-  /// shorthand for a 2D bin delta
-  using BinDelta2D = std::array<int, 2>;
+/// @brief Class that ties the digitization modules together and produces the channels
+class Channelizer {
+  PlanarSurfaceDrift m_surfaceDrift;
+  PlanarSurfaceMask m_surfaceMask;
+  Segmentizer m_segmentizer;
 
-  /// Nested struct for stepping from one channel to the next.
-  struct ChannelStep {
-    /// This is the delta to the last step in bins
-    BinDelta2D delta = {0, 0};
-    /// The intersection with the channel boundary
-    Acts::Vector2 intersect;
-    /// The patlength from the start
-    double path = 0.;
+ public:
+  /// Do the geometric channelizing
+  ///
+  /// @param hit The hit we want to channelize
+  /// @param surface the surface on which the hit is
+  /// @param gctx the Geometry context
+  /// @param driftDir the drift direction
+  /// @param segmentation the segmentation of the surface
+  /// @param thickness the thickness of the surface
+  ///
+  /// TODO Why having the thickness as a config parameter,
+  /// and not pulling it from the surface->detectorElement directly?
```

Comment:
Actually, we might want to remove the "thickness" from the detector element as it is a very planar detector oriented concept.